### PR TITLE
feat(Client): add deleteWebhook method

### PR DESCRIPTION
**Pull request by jaw0r3k**
**Please describe the changes this PR makes and why it should be merged:**

### <samp>🤖 Generated by Copilot at 3652f05</samp>

Add `deleteWebhook` method to `Client` class to enable deleting webhooks from Discord. Define `WebhookDeleteOptions` typedef for the method's options parameter.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/discord.js/src/client/Client.js [hunk 1]
```diff
@@ -253,6 +253,23 @@ class Client extends BaseClient {
     this.rest.setToken(null);
   }
 
+  /**
+   * Options used when deleting a webhook from Discord.
```
Comment (Jiralite) [inline packages/discord.js/src/client/Client.js:257]: The "from Discord" part is superfluous. Everything in this library is for Discord. No need to state "Create an event in Discord", "Delete a channel from Discord", etc.

```suggestion
   * Options used for deleting a webhook.
```

Comment (jaw0r3k) [reply packages/discord.js/src/client/Client.js]: Well every jsdoc here uses that
But yeah i agree

https://github.com/discordjs/discord.js/blob/5de608a576a0b390e45074bed16b8dd47710b38e/packages/discord.js/src/client/Client.js#L281

Comment (Jiralite) [reply packages/discord.js/src/client/Client.js]: We should move away from those patterns. I mean, that's what this library is for, after all.

**Review by Jiralite — COMMENTED** [review]
`Webhook#delete()` can call this method internally to avoid duplication:

https://github.com/discordjs/discord.js/blob/24fbb11ba2f7e8f8f604752159d2053f2cee16ec/packages/discord.js/src/structures/Webhook.js#L365-L375

**Review by sdanialraza — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

File: packages/discord.js/src/client/Client.js (hunk 1 above)
Comment (advaith1) [reply packages/discord.js/src/client/Client.js]: that's probably to clarify that it does an API request rather than using cache

**Pull request merged**